samples/bluetooth/peripheral_nucleo_wb55rg/src/main.c

```diff
@@ -0,0 +1,298 @@
+BT_GATT_SERVICE_DEFINE(stsensor_svc,
+BT_GATT_PRIMARY_SERVICE(&st_service_uuid),
+BT_GATT_CHARACTERISTIC(&led_char_uuid.uuid,
+	BT_GATT_CHRC_READ | BT_GATT_CHRC_WRITE_WITHOUT_RESP,
+	BT_GATT_PERM_WRITE, NULL, recv, (void *)1),
+BT_GATT_CHARACTERISTIC(&but_notif_uuid.uuid, BT_GATT_CHRC_NOTIFY,
+	BT_GATT_PERM_READ, NULL, NULL, &but_val),
+BT_GATT_CCC(mpu_ccc_cfg_changed,
+BT_GATT_PERM_READ | BT_GATT_PERM_WRITE),
+);
+
+static ssize_t recv(struct bt_conn *conn,
+		    const struct bt_gatt_attr *attr, const void *buf,
+		    u16_t len, u16_t offset, u8_t flags)
+{
+
+	if (led_dev) {
+		if (led_state == true) {
+			gpio_pin_write(led_dev, LED, 0);
+			LOG_INF("Turn off LED\n");
+		} else {
+			gpio_pin_write(led_dev, LED, 1);
+			LOG_INF("Turn on LED\n");
+		}
+		led_state = !led_state;
+	}
+
+	return 0;
+}
+
+void button_pressed(struct device *gpiob, struct gpio_callback *cb,
+		    u32_t pins)
+{
+	int err;
+
+	LOG_INF("Button SW1 pressed \n");
+
+	if (p_conn) {
+
+		if (notifyEnable == true) {
+
+			err = bt_gatt_notify(NULL, &stsensor_svc.attrs[2],
+			(const void *)&but_val, sizeof(u16_t));
+			if (err) {
+				LOG_ERR("Notify err\n");
+			} else {
+				LOG_INF("Send notify ok\n");
+				but_val = (but_val == 0) ? 0x100 : 0;
+			}
+
+		} else {
+			LOG_INF("Notify not enabled \n");
+		}
+
+	} else {
+		LOG_INF("BLE not connected\n");
+	}
+
+}
+
+void button_init(void)
+{
+	button_dev = device_get_binding(PORT);
+	if (!button_dev) {
+		LOG_ERR("Button error\n");
+		return;
+	}
+
+	gpio_pin_configure(button_dev, PIN,
+			   (GPIO_DIR_IN | GPIO_INT | GPIO_INT_EDGE |
+			    GPIO_PUD_PULL_UP |
+			    GPIO_INT_DEBOUNCE | GPIO_INT_ACTIVE_LOW));
+	gpio_init_callback(&gpio_cb, button_pressed, BIT(PIN));
+
+	gpio_add_callback(button_dev, &gpio_cb);
+	gpio_pin_enable_callback(button_dev, PIN);
+	but_val = 0;
+}
+
+void led_init(void)
+{
+
+	led_dev = device_get_binding(LED_PORT);
+	if (!led_dev) {
+		LOG_ERR("LED dev error\n");
+		return;
+	}
+	/* Set LED pin as output */
+	gpio_pin_configure(led_dev, LED, GPIO_DIR_OUT);
+	led_state = false;
+	gpio_pin_write(led_dev, LED, 0);
+}
+
+
+static void bt_ready(int err)
+{
+
+	if (err) {
+		printk("Bluetooth init failed (err %d)\n", err);
+		return;
+	}
+
+	LOG_INF("Bluetooth initialized\n");
+
+	/* Start advertising */
+	err = bt_le_adv_start(BT_LE_ADV_CONN, ad, ARRAY_SIZE(ad), NULL, 0);
+
+	if (err) {
+		LOG_ERR("Advertising failed to start (err %d)\n", err);
+		return;
+	}
+	p_conn = NULL;
+	LOG_INF("Configuration mode: waiting connections...\n");
+
+}
+
+
+static void connected(struct bt_conn *conn, u8_t err)
+{
+	if (err) {
+		LOG_ERR("Connection failed (err %u)\n", err);
+	} else {
+		LOG_INF("Connected\n");
+		p_conn = conn;
+	}
+}
+
+static void disconnected(struct bt_conn *conn, u8_t reason)
+{
+	p_conn = NULL;
+	LOG_INF("Disconnected (reason %u)\n", reason);
+
+}
+
+static struct bt_conn_cb conn_callbacks = {
+	.connected = connected,
+	.disconnected = disconnected,
+};
+
+static void bas_notify(void)
```

Comment:
Haven't been able to get this service working (tested on disco_l475_iot1)

Reply:
On Android ST BLE Sensor app the battery service is not working.
Tested on 4.4.4 version .Always showing "Battery not available"
Even using original STM32CubeIDE p2psrv example.
Android app issue? I will check on ST forum

Reply:
> On Android ST BLE Sensor app the battery service is not working.

Well, then we should remove this part. We can keep the placeholder thought for future use if that makes sense.

Reply:
battery service removed